src/main/java/org/springframework/test/context/junit5/external/ExternalApplicationContext.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2015-2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.test.context.junit5.external;
+
+import org.junit.gen5.api.extension.ExtendWith;
+import org.springframework.context.ApplicationContext;
+import org.springframework.core.annotation.AliasFor;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Resolve dependencies on annotated test class from {@link ApplicationContext} provided by {@link ApplicationContextProvider}.
+ *
+ * This annotation is a composed annotation.
+ *
+ * @author Tadaya Tsuyukubo
+ * @since 5.0
+ */
+@ExtendWith(SpringInjectionExtension.class)
+@Documented
+@Inherited
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ ElementType.TYPE })
+public @interface ExternalApplicationContext {
+
+	@AliasFor("provider")
+	Class<? extends ApplicationContextProvider> value() default ApplicationContextProvider.class;
+
+	@AliasFor("value")
+	Class<? extends ApplicationContextProvider> provider() default ApplicationContextProvider.class;
```

Comment:
Tip: I wouldn't introduce an aliased attribute if there is only one logical attribute to begin with.